packages/trader/src/AppV2/Components/Carousel/carousel.tsx

```diff
@@ -8,26 +8,32 @@ type THeaderProps = {
 type TCarousel = {
     header: ({ current_index, onNextClick, onPrevClick }: THeaderProps) => JSX.Element;
     pages: { id: number; component: JSX.Element }[];
+    current_index?: number;
+    setCurrentIndex?: (index: number) => void;
 };
 
-const Carousel = ({ header, pages }: TCarousel) => {
-    const [current_index, setCurrentIndex] = React.useState(0);
+const Carousel = ({ header, pages, current_index, setCurrentIndex }: TCarousel) => {
+    const [internalIndex, setInternalIndex] = useState(0);
 
-    const HeaderComponent = header;
+    const isControlled = current_index !== undefined && setCurrentIndex !== undefined;
```

Comment:
I need the internal page index outside the component so added a logic where if I pass state that it sets the external state or else it'll rely on internal State